WeakAuras/RegionTypes/Text.lua

```diff
@@ -169,76 +174,124 @@ local function modify(parent, region, data)
     end
   end
 
-  local UpdateText
-  if Private.ContainsAnyPlaceHolders(data.displayText) then
-    local getter = function(key, default)
-      local fullKey = "displayText_format_" .. key
-      if (data[fullKey] == nil) then
-        data[fullKey] = default
+  local containsCustomText = false
+  if Private.ContainsCustomPlaceHolder(data.displayText) then
+    containsCustomText = true
+  end
+
+  if not containsCustomText and type(data.conditions) == "table" then
+    for _, condition in ipairs(data.conditions) do
+      if type(condition.changes) == "table" then
+        for _, change in ipairs(condition.changes) do
+          if type(change.property) == "string"
+          and change.property == "displayText"
+          and type(change.value) == "string"
+          and Private.ContainsCustomPlaceHolder(change.value)
+          then
+            containsCustomText = true
+            break
+          end
+        end
       end
-      return data[fullKey]
     end
-    local formatters = Private.CreateFormatters(data.displayText, getter)
-    UpdateText = function()
-      local textStr = data.displayText;
-      textStr = Private.ReplacePlaceHolders(textStr, region, nil, false, formatters);
-      if (textStr == nil or textStr == "") then
-        textStr = " ";
+  end
+
+  if containsCustomText and data.customText and data.customText ~= "" then
+    data.customTextFunc = WeakAuras.LoadFunction("return "..data.customText)
+  else
+    data.customTextFunc = nil
+  end
+
+  function region:ConfigureTextUpdate()
+    local UpdateText
+    if self.displayText and Private.ContainsAnyPlaceHolders(self.displayText) then
+      local getter = function(key, default)
+        local fullKey = "displayText_format_" .. key
+        if (data[fullKey] == nil) then
+          data[fullKey] = default
+        end
+        return data[fullKey]
       end
+      local formatters = Private.CreateFormatters(self.displayText, getter)
+      UpdateText = function()
+        local textStr = self.displayText;
+        textStr = Private.ReplacePlaceHolders(textStr, self, nil, false, formatters);
+        if (textStr == nil or textStr == "") then
+          textStr = " ";
+        end
 
-      SetText(textStr)
+        SetText(textStr)
+      end
     end
-  end
 
-  local customTextFunc = nil
-  if(Private.ContainsCustomPlaceHolder(data.displayText) and data.customText) then
-    customTextFunc = WeakAuras.LoadFunction("return "..data.customText)
-  end
+    local customTextFunc = nil
+    if(Private.ContainsCustomPlaceHolder(self.displayText) and data.customText) then
+      customTextFunc = WeakAuras.LoadFunction("return "..data.customText)
```

Comment:
That shouldn't be done in ConfigureTextUpdatem since the function should be already loaded.